sdk/samples/iot/paho_iot_pnp_with_provisioning_sample.c

```diff
@@ -242,10 +243,32 @@ static void register_device_with_provisioning_service(void)
     exit(rc);
   }
 
+  // Devices registering a ModelId while using Device Provisioning Service must specify
+  // their ModelId in an MQTT payload sent during registration.
+  uint8_t mqtt_payload[MQTT_PAYLOAD_BUFFER_LENGTH];
+  size_t mqtt_payload_length;
+
+  az_span custom_payload_property
+      = AZ_SPAN_FROM_STR("{\"modelId\":\"dtmi:com:example:Thermostat;1\"}");
```

Comment:
only nit here is to maybe have the string as a #define at the top

Reply:
OK - small change, I went ahead and merged this to get in asap with hope I can get a PR out for the pnp=>main asap.  I don't expect that to be reviewed tonight of course as it grows late, but I at least hope to get a gate run so I can get eyes on it tomorrow.

Once we get the "core" changes in for a Thursday night run on the core stuff, I'll be able to address feedback like this and some other more minor documentation cleanup that's not as urgent.